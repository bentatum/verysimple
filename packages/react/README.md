# @verysimple/react

This is the `@verysimple/react` package