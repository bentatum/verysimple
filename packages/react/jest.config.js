/** @type {import('jest').Config} */
module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 72,
      functions: 75,
      lines: 75,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
  prettierPath: null,
};
