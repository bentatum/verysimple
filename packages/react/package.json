{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
