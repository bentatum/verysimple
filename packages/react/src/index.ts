export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
