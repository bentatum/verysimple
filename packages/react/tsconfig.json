{
  "extends": "@verysimple/tsconfig/react-library.json",
  "compilerOptions": {
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
