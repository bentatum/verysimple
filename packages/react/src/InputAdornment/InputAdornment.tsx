import { fieldPaddingClassNames } from "@/helpers/fieldPaddingClassNames";
import { InputSize } from "@/types";
import classNames from "classnames";
import { ComponentPropsWithRef, FC, ElementType, forwardRef } from "react";

export interface InputAdornmentProps extends ComponentPropsWithRef<"div"> {
  as?: ElementType;
  direction: "left" | "right";
  inputSize?: InputSize;
}

export const InputAdornment: FC<InputAdornmentProps> = forwardRef<
  HTMLDivElement,
  InputAdornmentProps
>(
  (
    { as, children, inputSize = "md", direction, className = "", ...props },
    ref
  ) => {
    const Component = as || "div";
    return (
      <Component
        ref={ref}
        className={classNames(
          "absolute inset-y-0",
          "flex items-center",
          "font-medium whitespace-nowrap",
          fieldPaddingClassNames(inputSize, className),
          {
            "pointer-events-none": !className.match(/pointer-events-/),
            "text-zinc-500 dark:text-zinc-300": !className.match(/text-/),
            "rounded-l-xl -left-px": direction === "left",
            "rounded-r-xl -right-px": direction === "right",
          },
          className
        )}
        {...props}
      >
        {children}
      </Component>
    );
  }
);

InputAdornment.displayName = "InputAdornment";

export default InputAdornment;
