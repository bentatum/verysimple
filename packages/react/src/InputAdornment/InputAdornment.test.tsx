import { render } from "@testing-library/react";
import InputAdornment from "../InputAdornment";

describe("InputAdornment", () => {
  it("should match the snapshot", () => {
    const { container } = render(<InputAdornment direction="left" />);
    expect(container).toMatchSnapshot();
  });

  it("renders children correctly", () => {
    const { getByText } = render(<InputAdornment direction="left">Test</InputAdornment>);
    expect(getByText("Test")).toBeInTheDocument();
  });
});
