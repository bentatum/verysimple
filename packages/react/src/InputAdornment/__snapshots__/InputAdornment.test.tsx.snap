// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdornment should match the snapshot 1`] = `
<div>
  <div
    class="absolute inset-y-0 flex items-center font-medium whitespace-nowrap px-5 pointer-events-none text-zinc-500 dark:text-zinc-300 rounded-l-xl -left-px"
  />
</div>
`;
